index.d.ts

```diff
@@ -0,0 +1,111 @@
+export interface Action<T> {
+  type: T;
+}
+
+
+/* reducers */
+
+export type Reducer<S> = <A extends Action<any>>(state: S, action: A) => S;
+
+export interface ReducersMapObject {
+  [key: string]: Reducer<any>;
+}
+
+export function combineReducers<S>(reducers: ReducersMapObject): Reducer<S>;
+export function combineReducers<S, M extends ReducersMapObject>(
+  reducers: M
+): Reducer<S>;
+
+
+/* store */
+
+export interface Dispatch {
+  <A>(action: A): A;
+  <A, B>(action: A): B;
+}
+
+export interface Unsubscribe {
+  (): void;
+}
+
+export interface Store<S> {
+  dispatch: Dispatch;
+  getState(): S;
+  subscribe(listener: () => void): Unsubscribe;
+  replaceReducer(reducer: Reducer<S>): void;
+}
+
+export interface StoreCreator {
+  <S>(reducer: Reducer<S>): Store<S>;
+  <S>(reducer: Reducer<S>, initialState: S): Store<S>;
+
+  <S>(reducer: Reducer<S>, enhancer: StoreEnhancer): Store<S>;
+  <S>(reducer: Reducer<S>, initialState: S, enhancer: StoreEnhancer): Store<S>;
```

Comment:
These can be simplified to
```typescript
export interface StoreCreator {
  <S>(reducer: Reducer<S>, enhancer?: StoreEnhancer): Store<S>;
  <S>(reducer: Reducer<S>, initialState: S, enhancer?: StoreEnhancer): Store<S>;
}
```
while still providing the same type information and it better describes the actual API.